crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java

```diff
@@ -0,0 +1,139 @@
+import uk.gov.justice.laa.crime.meansassessment.dto.maatcourtdata.HardshipReviewDTO;
+import uk.gov.justice.laa.crime.meansassessment.dto.maatcourtdata.IOJAppealDTO;
+import uk.gov.justice.laa.crime.meansassessment.dto.maatcourtdata.PassportAssessmentDTO;
+import uk.gov.justice.laa.crime.meansassessment.model.common.ApiAssessmentSummary;
+import uk.gov.justice.laa.crime.meansassessment.model.common.ApiCreateMeansAssessmentResponse;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.AssessmentType;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.CurrentStatus;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.ReviewType;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.WorkType;
+
+import java.util.ArrayList;
+
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AssessmentSummaryService {
+
+    private final MaatCourtDataService maatCourtDataService;
+
+    public ApiCreateMeansAssessmentResponse addAssessmentSummaryToMeansResponse(final ApiCreateMeansAssessmentResponse assessmentResponse,
+                                                                                final String laaTransactionId) {
+        log.info("Generating assessment summary for means assessment response");
+        try {
+            var assesmentsSummary = new ArrayList<ApiAssessmentSummary>();
+            assesmentsSummary.add(getFinancialAssessmentSummary(assessmentResponse));
+            assesmentsSummary.add(getPassportAssessmentSummary(assessmentResponse.getRepId(), laaTransactionId));
+            assesmentsSummary.add(getHardshipReviewsSummary(assessmentResponse.getRepId(), laaTransactionId));
+            assesmentsSummary.add(getIOJAppealSummary(assessmentResponse.getRepId(), laaTransactionId));
+            assessmentResponse.setAssessmentSummary(assesmentsSummary);
+        } catch (Exception ex) {
+            log.error("Failed to generate assessment summary for means assessment response with assessmentId: {}", assessmentResponse.getAssessmentId(), ex);
+        }
+        return assessmentResponse;
+    }
+
+    private ApiAssessmentSummary getFinancialAssessmentSummary(final ApiCreateMeansAssessmentResponse assessmentResponse) {
+        log.info("Generating assessment summary for financial assessment");
+        try {
+            ApiAssessmentSummary finAssessmentSummary = new ApiAssessmentSummary();
+            finAssessmentSummary.withId(assessmentResponse.getAssessmentId())
+                    .withReviewType(ofNullable(assessmentResponse.getReviewType()).map(ReviewType::getCode).orElse(null));
+
+            if (assessmentResponse.getAssessmentType() == AssessmentType.INIT) {
+                finAssessmentSummary.setType(WorkType.Initial_Assessment);
+                finAssessmentSummary.setStatus(ofNullable(assessmentResponse.getFassInitStatus()).map(CurrentStatus::getDescription)
```

Comment:
Can the `setStatus` call be brought outside of the `if/else`?

Reply:
No, condition is in place to set InitStatus and FullStatus